Source/Core/Core/GeckoCodeConfig.cpp

```diff
@@ -22,8 +23,23 @@ std::vector<GeckoCode> DownloadCodes(std::string gametdb_id, bool* succeeded, bo
   // TODO: Fix https://bugs.dolphin-emu.org/issues/11772 so we don't need this workaround
   const std::string protocol = use_https ? "https://" : "http://";
 
-  // codes.rc24.xyz is a mirror of the now defunct geckocodes.org.
-  std::string endpoint{protocol + "codes.rc24.xyz/txt.php?txt=" + gametdb_id};
+  std::string endpoint;
+
+  std::string GeckoURL = Config::GetBase(Config::MAIN_GECKO_URL);
+  if (GeckoURL.find("https://") != std::string::npos)
+  {
+    GeckoURL.erase(0, 8);
+  }
+
+  if (GeckoURL != "")
+  {
+    endpoint = protocol + GeckoURL + gametdb_id;
```

Comment:
This risks adding a double protocol if the setting specifies http://. The logic feels rather fragile.

Reply:
True, I could another if to check for http://

Reply:
Adding a check for protocols in general should be simpler and more reliable. Unless we want to handle http:// specially and not forcibly turn it into https://?

Reply:
Would it best if I add a check like https:// to remove the string, but for http://

Reply:
> Adding a check for protocols in general should be simpler and more reliable. Unless we want to handle http:// specially and not forcibly turn it into https://?

sorry github didn't update comments, that sounds like it would be more reliable. I could look into doing that

Reply:
dolphin appears to send an http request and not https

Reply:
It is always an HTTP request, HTTPS just specifies it is encrypted in transit.

Reply:
ok